### PR TITLE
Minor typescript cleanups

### PR DESCRIPTION
**Pull request by lindner**
- Use readonly where available
- Add component as an optional initializer for id.ts
- mark some classes protected

### CODE REVIEW

File: runtime/ts/arc.ts [hunk 1]
```diff
@@ -46,27 +46,27 @@ type PlanCallback = (recipe: Recipe) => void;
 type SerializeContext = {handles: string, resources: string, interfaces: string, dataResources: Map<string, string>};
 
 export class Arc {
-  private _context: Manifest;
-  private pecFactory: (id: string) => PECInnerPort;
-  private speculative: boolean;
+  private readonly _context: Manifest;
```
Comment (shaper) [inline runtime/ts/arc.ts]: Why do we still have some private member data with `_` prefix, are these just stragglers?

Comment (shans) [reply runtime/ts/arc.ts]: If the private member is shadowed by a getter (i.e. readonly, except from the owning class) then we encode that with a `_` prefix. Not sure if that's the case here.

File: runtime/ts/manifest.ts [hunk 2]
```diff
@@ -115,7 +115,7 @@ export class Manifest {
   storeTags: Map<StorageProviderBase, string[]> = new Map();
   private _fileName: string|null = null;
   private nextLocalID = 0;
-  private _id: Id;
+  private readonly _id: Id;
```
Comment (shans) [inline runtime/ts/manifest.ts]: consider converting this to `public readonly id: Id;` with no getter.

File: runtime/ts/particle-spec.ts [hunk 3]
```diff
@@ -110,7 +110,7 @@ export class ProvidedSlotSpec {
 type SerializedParticleSpec = {name: string, id?: string, verbs: string[], args: SerializedConnectionSpec[], description: {pattern?: string}, implFile: string, affordance: string[], slots: SerializedSlotSpec[]};
 
 export class ParticleSpec {
-  private _model: SerializedParticleSpec;
+  private readonly _model: SerializedParticleSpec;
```
Comment (shans) [inline runtime/ts/particle-spec.ts]: ditto, and for other instances below of `private readonly _member`

File: runtime/ts/storage/pouchdb/pouch-db-storage-provider.ts [hunk 4]
```diff
@@ -23,7 +23,7 @@ export abstract class PouchDbStorageProvider extends StorageProviderBase {
   /** The Pouch revision of the data we have stored locally */
   protected _rev: string | undefined;
 
-  constructor(type: Type, storageEngine: PouchDbStorage, name: string, id: string, key: string) {
+  protected constructor(type: Type, storageEngine: PouchDbStorage, name: string, id: string, key: string) {
```
Comment (shans) [inline runtime/ts/storage/pouchdb/pouch-db-storage-provider.ts]: Does `protected` for constructors imply that the class can only be constructed by itself or a subclass?

File: runtime/ts/manifest.ts (hunk 2 above)
Comment (lindner) [reply runtime/ts/manifest.ts]: Will look into a followup on this.  There's some logic in the getter:

```
  get id() {
    if (this._meta.name) {
      return Id.newSessionId().fromString(this._meta.name);
    }
    return this._id;
  }
```

**Pull request merged**